assignment-1/IPL.java

```diff
@@ -0,0 +1,329 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.util.*;
+
+import static java.util.Map.Entry.comparingByValue;
+import static java.util.stream.Collectors.toMap;
+
+
+public class IPL {
+
+    static String[] teamNames = {"CSK", "MI", "RR", "RCB", "SRH", "PBKS", "DC", "KKR"};
+    static String[] venueNames = {"CHENNAI", "MUMBAI", "JAIPUR", "BENGALURU", "HYDERABAD", "CHANDIGARH", "DELHI", "KOLKATA"};
+    static ArrayList<Team> teams = new ArrayList<>();
+    static ArrayList<Match> matches = new ArrayList<>();
+
+    public static void fortyWicketsClub(String teamName) {
+        for (Team team : teams) {
+            if (team.teamName.equals(teamName)) {
+                for (Player p : team.squad) {
+                    if (p.wickets > 40) {
+                        p.display();
+                    }
+                }
+            }
+        }
+    }
+
+    public static void searchPlayersByName(String key) {
+        for (Team team : teams) {
+            for (Player p : team.squad) {
+                if (p.playerName.contains(key)) {
+                    p.display();
```

Comment:
the param key could have a better name
You also need to handle when the input param is null
A custom exception saying that no such player exists was expected
Player player is better than Player p